File: src/alarme_termico.cpp
#include "alarme_termico.hpp"

void AlarmeTermico::avaliar(double temperatura) {

    if (temperatura > 45.0) {
        ligado_ = true;
    } else if (temperatura < 40.0) {
        ligado_ = false;
    }

}

bool AlarmeTermico::estaLigado() const { return ligado_; }

File: src/alarme_termico.py
class AlarmeTermico:
    def __init__(self) -> None:
        self._ligado = False

    def avaliar(self, temperatura: float) -> None:
        if temperatura > 45.0:
            self._ligado = True
        elif temperatura < 40.0:
            self._ligado = False

    @property
    def ligado(self) -> bool:
        return self._ligado